lib/perl/Genome/Model/CwlPipeline/Command/Cleanup.pm

```diff
@@ -0,0 +1,62 @@
+package Genome::Model::CwlPipeline::Command::Cleanup;
+
+use strict;
+use warnings;
+
+use File::Spec;
+use Genome;
+
+use Genome::Utility::File::Mode qw(mode);
+
+class Genome::Model::CwlPipeline::Command::Cleanup {
+    is => 'Command::V2',
+    has => [
+        builds => {
+            is => 'Genome::Model::Build::CwlPipeline',
+            is_many => 1,
+            doc => 'The build(s) to cleanup disk.',
+        },
+    ],
+    doc => 'cleanup disk for older CwlPipeline builds.',
+};
+
+sub help_detail {
+    return <<EOHELP
+Remove the tmp* directories and the content of those directories from the build data directory. Also reallocate the size after removal.
+EOHELP
+    ;
+}
+
+sub execute {
+    my $self = shift;
+
+    for my $build ($self->builds) {
+        unless ($build->status eq 'Succeeded') {
+            $self->fatal_message("Unable to run cleanup on '%s' build with id '%s'. For Failed builds, please abandon instead once troubleshooting is complete.",$build->status,$build->id);
+        }
+
+        my $data_directory = $build->data_directory;
+
+        my $results_dir = File::Spec->join($data_directory, 'results');
+
+        my $tmp_dir = Genome::Sys->create_temp_directory($build->id);
+        my $results_dir_mode = mode($results_dir);
+        $results_dir_mode->add_user_writable();
+        unless (Genome::Model::CwlPipeline::Command::Run->cleanup($tmp_dir, $results_dir)) {
+            $self->fatal_message("Failed to cleanup build tmp dir '%s' and results dir '%s'", $tmp_dir, $results_dir);
```

Comment:
Failure here will leave the permissions loosened on the directory!

Reply:
I assume I want to have some kind of scope guard or use an eval to cleanup a failed call. Suggestions?
```
eval { cleanup() }
if ($@) { $results_dir_mode->rm_user_writable(); $self->fatal_message();}
```

Reply:
A `Scope::Guard` is probably good enough.  Otherwise I'd suggest `Try::Tiny` for the `finally`.